sklearn/preprocessing/knn_imputation.py

```diff
@@ -0,0 +1,253 @@
+# Authors: Ashim Bhattarai <ashimb9@gmail.com>
+# License: BSD 3 clause
+
+from __future__ import division
+import warnings
+import numpy as np
+
+from ..base import BaseEstimator, TransformerMixin
+from ..utils import check_array
+from ..utils.validation import check_is_fitted
+from ..utils.validation import FLOAT_DTYPES
+from ..neighbors import NearestNeighbors
+from ..neighbors.base import _get_weights, _check_weights
+__all__ = [
+    'KNNImputer',
+]
+
+
+def _get_mask(X, value_to_mask):
+    """Compute the boolean mask X == missing_values."""
+    if value_to_mask == "NaN" or np.isnan(value_to_mask):
+        return np.isnan(X)
+    else:
+        return X == value_to_mask
+
+
+class KNNImputer(BaseEstimator, TransformerMixin):
+    """Imputation transformer for completing missing values using Nearest
+    Neighbors. Broadly speaking, the imputation is performed using either
+    the weighted or the unweighted mean of the desired number of neighbors.
+
+    Parameters
+    ----------
+    missing_values : integer or "NaN", optional (default="NaN")
+        The placeholder for the missing values. All occurrences of
+        `missing_values` will be imputed. For missing values encoded as np.nan,
+        use the string value "NaN".
+
+    n_neighbors : int, optional (default = 5)
+        Number of neighbors to get.
```

Comment:
"Maximum number of neighboring samples to use for imputation. When the feature being imputed is not available ..."